### PR TITLE
Transferfunction fixes

### PR DESCRIPTION
**Pull request by petersteneteg**
Use reference instead of pointers in the interfaces.

solves issue with deserialization of tfs...

### CODE REVIEW

File: modules/qtwidgets/src/tf/tfeditorprimitive.cpp [hunk 1]
```diff
@@ -76,21 +76,21 @@ TFEditorPrimitive::TFEditorPrimitive(TFPrimitive* primitive, QGraphicsScene* sce
     }
 }
 
-void TFEditorPrimitive::setPrimitive(TFPrimitive* primitive) { data_ = primitive; }
+TFPrimitive& TFEditorPrimitive::getPrimitive() { return data_; }
```
Comment (martinfalk) [inline modules/qtwidgets/src/tf/tfeditorprimitive.cpp:79]: what happened to `set()`?

File: src/core/datastructures/tfprimitiveset.cpp [hunk 2]
```diff
@@ -326,77 +360,63 @@ void TFPrimitiveSet::load(const std::string&, const FileExtension&) {}
...
-        for (size_t i = 0; i < size; i++) {
-            const auto val = static_cast<float>(i) / (size - 1);
-            dataArray[i] = vec4(val, val, val, 1.0);
-        }
-    } else if (sorted_.size() == 1) {  // in case of 1 point
-        const auto color = sorted_.front()->getColor();
-        for (size_t i = 0; i < size; ++i) {
-            dataArray[i] = color;
-        }
+    if (size() == 0) {  // in case of 0 points
```
Comment (martinfalk) [inline src/core/datastructures/tfprimitiveset.cpp]: empty() ?

**Review by martinfalk — APPROVED** [review]

File: modules/qtwidgets/src/tf/tfeditorprimitive.cpp (hunk 1 above)
Comment (petersteneteg) [reply modules/qtwidgets/src/tf/tfeditorprimitive.cpp:79]: can't reset a reference...

File: src/core/datastructures/tfprimitiveset.cpp (hunk 2 above)
Comment (petersteneteg) [reply src/core/datastructures/tfprimitiveset.cpp]: true

**Pull request merged**